---
title: "MutationType"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{MutationType}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r style, echo = FALSE, results = 'asis'}
library(BiocStyle)
```

```{r, echo = FALSE}
library(knitr)
```


```{r setup, results="hide", include = FALSE}
library(MutationType)
```

Usage

First, the VariantAnnotation, BSgenome and a specific BSgenome libraries (e.BSgenome.Hsapiens.UCSC.hg19) should be loaded.

```{r}
library("VariantAnnotation")
library("BSgenome")
library("BSgenome.Hsapiens.UCSC.hg19")
```

Then, if you do not have your own VCF file, you can load an example one from Variant Annotation package. A VCF file can be loaded as a CollapsedVCF using the 'readVcf' function from the 'VariantAnnotation' package with a reference genome from BSgenome library (e.g "hg19").

```{r}
vcffile <- system.file("extdata", "chr22.vcf.gz", package="VariantAnnotation")
vcf <- readVcf(vcffile, "hg19")
```

Finally, you can extract SNV mutations from VCF file using determine_mutation_types function by calling it. It requires vcf file as CollapsedVCF class, reference genome from specific BSgenome library (e.g BSgenome.Hsapiens.UCSC.hg19) and context length, which should be odd positive integer. 

```{r}
determine_mutation_types(vcf, Hsapiens, 3)
```

If you want to count the number of each SNV mutation type, you can use "count_SNVmutations" function, which takes a vcf file and reference genome and returns a count table for each SNV mutation type and a graphical representation of it. 

```{r}
count_SNVmutations(vcf, Hsapiens)
```

